src/OpenApi/JSON.hs

```diff
@@ -0,0 +1,87 @@
+module OpenApi.JSON
+  ( generateRenamed
+  , parseJSONFixed
+  , parseRefSum
+  , parseRenamed
+  )
+where
+
+import Data.List (lookup)
+import Data.String (String)
+import GHC.Enum (Bounded, Enum, minBound)
+import GHC.Generics (Generic, Rep)
+import OpenApi.Prelude
+
+import qualified Data.Aeson.Types    as JSON
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.Text           as Text
+
+parseRenamed
+  :: (Generic a, JSON.GFromJSON JSON.Zero (Rep a))
+  => [(String, String)]
```

Comment:
Could we assert this type is a Map instead of a list of pairs? I assume the "key" is unique, because it wouldn't make sense for the same key to be present more than once?

Reply:
Yeah, thats a improvement I always wanted to do but delayed as my mind was on the consumer of this library. This is the right moment to do it.